libkbfs/tlf_handle.go

```diff
@@ -931,40 +919,78 @@ func ParseTlfHandle(
 	if len(extensionSuffix) != 0 {
 		extensions, err = ParseTlfHandleExtensionSuffix(extensionSuffix)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 	}
 
 	h, err := makeTlfHandleHelper(ctx, public, writers, readers, extensions)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	if !public {
 		currentUsername, currentUID, err := kbpki.GetCurrentUserInfo(ctx)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
 		if !h.IsReader(currentUID) {
-			return nil, ReadAccessError{currentUsername, h.GetCanonicalName(), public}
+			return nil, nil, ReadAccessError{currentUsername, h.GetCanonicalName(), public}
 		}
 	}
 
-	if string(h.GetCanonicalName()) == name {
-		// Name is already canonical (i.e., all usernames and
-		// no assertions) so we can delay the identify until
-		// the node is actually used.
-		return h, nil
-	}
+	switch identifyBehavior {
+	case keybase1.TLFIdentifyBehavior_DEFAULT_KBFS:
+		if string(h.GetCanonicalName()) == name {
+			// Name is already canonical (i.e., all usernames and
+			// no assertions) so we can delay the identify until
+			// the node is actually used.
+			return h, nil, nil
+		}
 
-	// Otherwise, identify before returning the canonical name.
-	err = identifyHandle(ctx, kbpki, kbpki, h)
-	if err != nil {
-		return nil, err
+		// Otherwise, identify before returning the canonical name.
+		err = identifyHandle(ctx, kbpki, kbpki, h)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return nil, nil, TlfNameNotCanonical{name, string(h.GetCanonicalName())}
+	case keybase1.TLFIdentifyBehavior_CHAT:
+		breaks, err := identifyHandleWithIdentifyBehavior(ctx, kbpki, kbpki, h,
+			keybase1.TLFIdentifyBehavior_CHAT)
+		if err != nil {
+			return nil, breaks, err
+		}
+
+		if string(h.GetCanonicalName()) != name {
```

Comment:
Also, if KBFS has already done the full identify (e.g., `folderBranchOps.identifyDone = true`), this PR does extra work on every `GetTLFCryptKeys` RPC.